### ISSUE TITLE
Can't upload oci-image resources from any registry but Docker Hub

### ISSUE THREAD

**Issue by Cynerva** [canonical/charmcraft#350]
```
$ charmcraft upload-resource k8s-keystone-auth k8s-keystone-auth-image --image rocks.canonical.com/cdk/k8scloudprovider/k8s-keystone-auth:v1.21.0
The registry server cannot be specified as part of the image (the format is [organization/]name[:tag|@digest]). (full execution logs in /tmp/charmcraft-log-7qw9appt)

$ charmcraft version
1.0.0
```

My team has a variety of testing environments that do not have access to Docker Hub. Those environments do have access to rocks.canonical.com, though, and I would like to have the resource reference that registry instead. It seems that isn't allowed. Why?

**Comment by facundobatista** [comment]
We're working on the evolution of the current behaviour, because of that and other details.

In the near future, you will be able to just push images from your local machine (being the image just built there or brought from any other repository).

The specification of this new behaviour is [here](https://discourse.charmhub.io/t/charmcrafts-upload-resource-command/4580).

Thanks!

**Comment by facundobatista** [comment]
This is already released in 1.1. 

Please let me know if that solves your needs, or have a further question. Regards,
